updater/aws_test.go

```diff
@@ -15,6 +15,260 @@ import (
+		{
+			instanceID:          "inst-id-1",
+			containerInstanceID: "cont-inst-1",
+		},
+		{
+			instanceID:          "inst-id-2",
+			containerInstanceID: "cont-inst-2",
+		},
+		{
+			instanceID:          "inst-id-3",
+			containerInstanceID: "cont-inst-3",
+		},
+		{
+			instanceID:          "inst-id-4",
+			containerInstanceID: "cont-inst-4",
+		},
+		{
+			instanceID:          "inst-id-5",
+			containerInstanceID: "cont-inst-5",
+		},
+	}
+	expected := []instance{
+		{
+			instanceID:          "inst-id-1",
+			containerInstanceID: "cont-inst-1",
+			bottlerocketVersion: "v1.0.5",
+		},
+		{
+			instanceID:          "inst-id-2",
+			containerInstanceID: "cont-inst-2",
+			bottlerocketVersion: "v1.0.5",
+		},
+		{
+			instanceID:          "inst-id-5",
+			containerInstanceID: "cont-inst-5",
+			bottlerocketVersion: "v1.0.5",
+		},
+	}
+	responses := map[string]string{
+		"inst-id-1": "{\"update_state\": \"Available\", \"active_partition\": { \"image\": { \"version\": \"v1.0.5\"}}}",
+		"inst-id-2": "{\"update_state\": \"Ready\", \"active_partition\": { \"image\": { \"version\": \"v1.0.5\"}}}",
+		"inst-id-3": "{\"update_state\": \"Idle\", \"active_partition\": { \"image\": { \"version\": \"v1.1.1\"}}}",
+		"inst-id-4": "{\"update_state\": \"Staged\", \"active_partition\": { \"image\": { \"version\": \"v1.1.1\"}}}",
+		"inst-id-5": "{\"update_state\": \"Available\", \"active_partition\": { \"image\": { \"version\": \"v1.0.5\"}}}",
+	}
+	mockSSM := MockSSM{
+		GetCommandInvocationFn: func(input *ssm.GetCommandInvocationInput) (*ssm.GetCommandInvocationOutput, error) {
+			return &ssm.GetCommandInvocationOutput{
+				Status:                aws.String("Success"),
+				StandardOutputContent: aws.String(responses[*input.InstanceId]),
+			}, nil
+		},
+		SendCommandFn: func(input *ssm.SendCommandInput) (*ssm.SendCommandOutput, error) {
+			return &ssm.SendCommandOutput{
+				Command: &ssm.Command{
+					CommandId:    aws.String("command-id"),
+					DocumentName: aws.String("check-document"),
+				},
+			}, nil
+		},
+		WaitUntilCommandExecutedWithContextFn: func(ctx aws.Context, input *ssm.GetCommandInvocationInput, opts ...request.WaiterOption) error {
+			assert.Equal(t, "command-id", aws.StringValue(input.CommandId))
+			return nil
+		},
+	}
+	u := updater{ssm: mockSSM, checkDocument: "check-document"}
+	actual, err := u.filterAvailableUpdates(instances)
+	require.NoError(t, err)
+	assert.Equal(t, expected, actual, "Should only contain instances in Aavailable or Ready update state")
+}
+
+func TestPaginatedFilterAvailableUpdatesSuccess(t *testing.T) {
+	checkPattern := "{\"update_state\": \"%s\", \"active_partition\": { \"image\": { \"version\": \"%s\"}}}"
+	expected := make([]instance, 0)
+	instances := make([]instance, 0)
+	getOut := &ssm.GetCommandInvocationOutput{
+		Status:                aws.String("Success"),
+		StandardOutputContent: aws.String(fmt.Sprintf(checkPattern, updateStateAvailable, "v1.0.5")),
+	}
+
+	for i := 0; i < 100; i++ {
```

Comment:
nit: can you add a comment explaining 100 here?  I expect 100 is here because the SSM page size is 50 and this would be two full pages...